### PR TITLE
MAINT, API Change trunk-classification into three separate functions for generating trunk, trunk-mix, trunk-overlap and marron-wand

### PR DESCRIPTION
**Pull request by adam2392**
Changes proposed in this pull request:
- separates `make_trunk_classification` into three functions: `make_trunk_class*`, `make_trunk_mixture_class*`, and `make_marron_wand_class*` so that we can separate logic for simulating trunk, trunk-overlap, trunk-mix and marron-wand simulations

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: sktree/datasets/hyppo.py
Comment (adam2392) [inline sktree/datasets/hyppo.py]: @sampan501 can you explain why we have a (2/3)**2 here and only in trunk-mix?

Comment (sampan501) [reply sktree/datasets/hyppo.py]: When variance is 1, trunk-mix does not look bimodal at low dimensions. I set it to (2/3)**2 since that is consistent with Marron and Wand bimodal

**Review by sampan501 — APPROVED** [review]

**Review by YuxinB — APPROVED** [review]

**Comment by sampan501** [comment]
@adam2392 Good to merge after changelog update

File: sktree/datasets/hyppo.py
Comment (adam2392) [inline sktree/datasets/hyppo.py]: @sampan501 can you explain what the role of `w` is? I'm not sure why it's passed via `mu_1`, or if it should be?

Comment (YuxinB) [reply sktree/datasets/hyppo.py]: From this code, if I understand right, w should be the probability that we sample from G. Like (1-w)*N(0,1)+w*G, G is any Marron-Wand simulation.

**Review by sampan501 — CHANGES_REQUESTED** [review]
Looks good! For simplicity, there is a lot of error checks that are common for all 3 functions. Maybe we can wrap those up into a private method?

**Comment by sampan501** [comment]
Other than that, pending CI, lgtm

**Comment by adam2392** [comment]
> Looks good! For simplicity, there is a lot of error checks that are common for all 3 functions. Maybe we can wrap those up into a private method?

Good idea. Rn tho there's only a few.  We could migrate the covariance matrix stuff to a `_default_covariance(...)` which handles the check and setting of covariance possibly. But maybe we can leave for later depending on any future added complexity. Otw over-engineering maybe :p

**Comment by sampan501** [comment]
Fair, I'll make an issue for it

**Review by sampan501 — APPROVED** [review]

**Review by SUKI-O — APPROVED** [review]

**Pull request merged**
